android/BUILD.gn

```diff
@@ -29,5 +26,6 @@ java_cpp_template("brave_config_java") {
     "BRAVE_ANDROID_ENABLE_IPFS=$enable_ipfs",
     "BRAVE_ANDROID_WEBCOMPAT_REPORT_ENDPOINT=\"$webcompat_report_api_endpoint\"",
     "BRAVE_ANDROID_AI_CHAT_ENABLED=$enable_ai_chat",
+    "BRAVE_ANDROID_WEB_WALLET_ENABLED=$brave_android_web_wallet_enabled",
```

Comment:
that's not really a feature flag, it's a build flag. Feature flag is something that we can change dynamically without building a new binary and they are inside `brave://flags`. Do we want build flag or a feature flag that we can change without any rebuilds for that feature?

Reply:
That should be a feature flag indeed

Reply:
Apologies for using the wrong naming here, I've amended the PR description and changed it.
Build flag is fine, discussed also with James: we don't want our users to revert to the old native implementation as it will be removed completely in the future. 
We just needed a safety net to control and revert the behavior through a new build in case of unforeseen issues.

Reply:
> Build flag is fine, discussed also with James: we don't want our users to revert to the old native implementation as it will be removed completely in the future.
> We just needed a safety net to control and revert the behavior through a new build in case of unforeseen issues.

This is not for users, but to be controllable by griffin platform. How are we going to recover from possible disaster in production caused by this PR(which by its nature may affect non-wallet users also)? AFAIK that is not that simple for android to build a hotfix and deliver to everyone asap.

So as it not a trivial change to wallet we need a `base::Feature` toggle for this. After new ui is in release channel for a while(a month for example) and doing OK we can throw away this toggle and all related legacy code.

cc: @jamesmudgett @bbondy @yrliou

Reply:
I don't have much knowledge around the Griffin platform, but I'll be happy to implement the suggested solution if also the others feel the same, I might be biased here.

## Why I think it's overkill to implement a feature flag

We have no reason to believe a disaster in production may happen. This PR will be thoroughly reviewed and tested (as all our PRs), the worst case scenario will be having some small issues to fix. With James, we decided to introduce a build flag to quickly revert to the old native UI in case the new WebUI screens will be badly received from users, and we'll have a quick flag to modify.  At the beginning, the build flag was something we didn't even plan, we just thought to introduce it at a later stage to play safe as an extra precaution.

This PR should not affect non-wallet users, if you feel there are parts that may create side effects I'm happy to take a look as it's something that should not happen.

As I said, I might be biased here, as I've poured lots of hours into this PR, so I'm happy to hear further feedback from @jamesmudgett, @bbondy, and @yrliou and take the feature flag approach if they also feel the same.

Reply:
@srirambv what do you prefer for testing? With a feature flag you'd be able to switch between the two wallet implementations. However, I don't know that we'd need to hold this back unless we had a major crash or something impacting a large group of users.

Reply:
In my opinion we are fine without a feature flag if we are not going to uplift that work and it naturally migrates via channels.

Reply:
> With James, we decided to introduce a build flag to quickly revert to the old native UI in case the new WebUI screens will be badly received from users, and we'll have a quick flag to modify.

If you want a quick revert do this with a feature flag + griffin. Build flag it is not designed for a quick revert when there are already angry users

Reply:
> We have no reason to believe a disaster in production may happen. This PR will be thoroughly reviewed and tested (as all our PRs), 

Does this mean all disasters in production were expected beforehand? I'm sure not. That's why feature should exist to turn it off.


Guys, I don't understand how to explain this. 
There is literally 4K+ instances of `BASE_FEATURE` in the codebase. And it is a default way to introduce any new user visible change. Chromium invested many man-months to support easy development and deployment of new stuff which benefits both engineers and users with BASE_FEATURE and Finch(Griffin) working together. Why throw away all of these in favor of inventing shortliving buildflag?

I see two options. I'd prefer the second one:
1. If only this `We have no reason to believe a disaster in production may happen. This PR will be thoroughly reviewed and tested (as all our PRs), the worst case scenario will be having some small issues to fix.` then we neither need new feature nor buildflag and legacy code should be removed. If we need it back we always can revert PR.
2. If also this `we decided to introduce a build flag to quickly revert to the old native UI in case the new WebUI screens will be badly received from users, and we'll have a quick flag to modify.` then we need `base::Feature` and no buildflag.

Reply:
Just my two cents:

1. I think it takes about the same amount of time between toggling off build flag in production and reverting PRs. Both need uplifts, rebuild, and re-release. Between these two, build flag actually would add some complexity to the code base. So I don't think build flag is really needed if we just want at least a way to change things back. In this case, we can just ride the train and revert if it's bad that we decide to go back.
2. If we want a way to be able to quick rollback to native UI when needed without another release, then we should do feature flag + griffin. In general, feature flag is the one which should be used when we are developing non-trivial changes.

Personally I think 2 is more beneficial and flexible. But I'm okay if we decide to ride the train and revert when needed.

Reply:
What has happened is @simoarpe never implemented nor feature flag nor build flag. And made a build flag instead of a feature flag. Honestly I don't see why we would need a feature flag for that work at all, it's basically a UI re-work and is not going to be uplifted. Perhaps it would be a good idea to have a flag, but it's done that way. From that point doing a feature flag will delay that work and it's doubtful that it will bring any benefit as most likely we will never use it.

Reply:
Having said that I would just add a follow up issue to remove the build flag entirely when the feature hits stable channel.

Reply:
Do we usually do feature flags for desktop UI re-design? I don't think so.